packages/base/lib/plugin.js

```diff
@@ -73,9 +77,33 @@ export default class BasePlugin {
 
   /**
    * You could also handle all commands generically by implementing 'handle'
+   *
+   * @param {() => Promise<void>} next - A reference to an async function which
+   * encapsulates what would normally happen if this plugin were not handling a
+   * command.
+   * @param {BaseDriver} driver - the instance of the Appium driver currently
+   * handling commands
+   * @param {string} cmdName - the name of the command being handled
+   * @param {...object} args - the args that would be applied to the command
+   * @returns {Promise<any>}
    */
-  /*async handle (next, driver, cmdName, ...args) {
-     return await next();
-  }*/
+  // eslint-disable-next-line no-unused-vars
+  async handle (next, driver, cmdName, ...args) {
```

Comment:
we def don't want to uncomment this. it's designed as a convenience tool for plugins that expliclty want to handle every command. most plugins probably don't, and it makes everything much more complicated if every plugin handles every command.

Reply:
okay... except: the current state of affairs is kind of problematic in that the `BasePlugin` class does not describe what a `Plugin` can be.  We're going to have methods that have special meaning which are only defined in an external interface, but don't actually exist on the class itself.  And all of the `<cmd>` methods which are referenced in a comment: what methods are allowed?  What do they do?  This is presently undefined.

We shouldn't be conflating the class and the interface if the class does not _implement_ the interface, in other words.  I think we can avoid this by fully defining this interface and ensuring `BasePlugin` implements it.  unsure if there is any real technical reason why we can't do that.

* * *
> note: tangent follows

_personally_ I'd prefer a different API for extension authors, but I understand the desire to keep plugins/drivers the same and avoid breaking changes on the driver side.  in a nutshell, I'd describe my ideal API as: "extension authors provide a thing that implements an interface" and avoid worrying about classes or `this`; if an extension author wants to use a class which implements the interface, they can, but it's not required.  this would also simplify the declarations.